### ISSUE TITLE
Change default shell to bash

### ISSUE THREAD

**Issue by riker09** [w9jds/firebase-action#103]
My GitHub workflows broke over the weekend. A quick investigation showed that you updated your base image recently.

This is my GitHub action;

```yaml
      - name: Deploy to Firebase
        uses: w9jds/firebase-action@master
        with:
          args: deploy --only hosting:myapp --message \"${COMMIT_MESSAGE:0:200}\"
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
          PROJECT_ID: ${{ env.PROJECT_ID }}
```

and it results in this error: `sh: 1: Bad substitution`.

Yes, I know, relying on `@master` is a terrible idea and in fact the reason why *my* workflows went south. However, I would greatly appreciate it if you could change the default shell of the buster image back to `bash` so I could use `\"${COMMIT_MESSAGE:0:200}\".`

**Comment by w9jds** [comment]
This hasn't changed, it has used `sh` since the initial implementation. The issue was probably that it was pretty busted for about a day so you are right doing @master is a bad idea. The main reason for switching off alpine was mostly caused by issues with missing c libraries used by some things in functions.

**Comment by aravindvnair99** [comment]
@riker09 I hadn't changed the default shell. It is `sh` itself as @w9jds mentioned. But I just noticed something:

# On `alpine`

```console
$ which sh
/bin/sh
$ readlink -f $(which sh)
/bin/busybox
$ which bash
The command '/bin/sh -c which bash' returned a non-zero code: 1
$ readlink -f $(which bash)
BusyBox v1.31.1 () multi-call binary.

Usage: readlink [-fnv] FILE

Display the value of a symlink

        -f      Canonicalize by following all symlinks
        -n      Don't add newline
        -v      Verbose
The command '/bin/sh -c readlink -f $(which bash)' returned a non-zero code: 1
```

# On `buster`

```console
$ which sh
/bin/sh
$ readlink -f $(which sh)
/bin/dash
$ which bash
/bin/bash
$ readlink -f $(which bash)
/bin/bash
```

# TLDR

- `buster` has both `sh` and `bash` unlike `alpine` which only has `sh`
- `alpine` uses `busybox` whereas `buster` uses `dash`
- `sh` is the default, but both are symlinked to two different things

# Using `master`

And as for using the `master` branch, builds which failed such as my PRs #99 and #100 were not pushed to Docker Hub. But yes, `master` isn't recommended for any GitHub Actions unless you want to live on the "bleeding edge" as [releases are not released often especially in this repository](https://github.com/w9jds/firebase-action/releases).

# Why `buster` instead of `alpine`?

`alpine` lacks many tools/software/bundles which are required for various `npm` dependencies when using Google Cloud Functions. But they are present in `buster`. For example, TensorFlow which is widely used won't compile on `alpine` (#98). And although `alpine` is slightly bigger (1.6 times at the moment) than `buster`, GitHub Actions VMs are hooked up to high-speed internet and hence pulling images won't take time either. I only noticed a difference of 1 - 2.5 seconds in my test. (Stating this as users were concerned about time to deploy)

# Issue

The difference is in Debian dash (Debian Almquist shell) vs Busybox ash (Almquist shell). The substring parameter expansion is not part of the [POSIX shell specification](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html). Hence you can't do that in `dash`.

# Solution

I'm unsure at the moment although I do see this: https://stackoverflow.com/questions/47332006/substring-in-sh-returns-bad-substitution. Will follow up if I have any suggestions or workarounds.
